demisto_sdk/commands/create_id_set/tests/update_id_set_test.py

```diff
@@ -147,7 +149,131 @@ def test_create_id_set_on_specific_empty_pack(self, repo):
         with open(self.file_path, 'r') as id_set_file:
             private_id_set = json.load(id_set_file)
         for content_entity, content_entity_value_list in private_id_set.items():
-            assert len(content_entity_value_list) == 0
+            if content_entity != 'Packs':
+                assert len(content_entity_value_list) == 0
+            else:
+                assert len(content_entity_value_list) == 1
+
+
+class TestPacksMetadata:
+    METADATA_WITH_XSOAR_SUPPORT = {
+        'name': 'Pack1',
+        'support': 'xsoar',
+        'currentVersion': '1.0.0',
+        'author': 'Cortex XSOAR',
+        'tags': ['Alerts'],
+        'useCases': ['Case Management'],
+        'categories': ['Endpoint']
+    }
+
+    METADATA_WITH_PARTNER_SUPPORT = {
+        'name': 'Pack1',
+        'support': 'partner',
+        'currentVersion': '1.0.0',
+        'author': 'Some Partner',
+        'tags': ['Alerts'],
+        'useCases': ['Case Management'],
+        'categories': ['Endpoint']
+    }
+
+    METADATA_WITH_COMMUNITY_SUPPORT = {
+        'name': 'Pack1',
+        'support': 'community',
+        'currentVersion': '1.0.0',
+        'author': 'Someone',
+        'tags': ['Alerts'],
+        'useCases': ['Case Management'],
+        'categories': ['Endpoint']
+    }
+
+    TEST_PACK = [
+        (METADATA_WITH_XSOAR_SUPPORT, 'Cortex XSOAR', 'certified'),
+        (METADATA_WITH_PARTNER_SUPPORT, 'Some Partner', 'certified'),
+        (METADATA_WITH_COMMUNITY_SUPPORT, 'Someone', ''),
+    ]
+
+    @staticmethod
+    @pytest.mark.parametrize('metadata_file_content, author, certification', TEST_PACK)
+    def test_process_metadata(repo, metadata_file_content, author, certification):
+        """
+        Given
+            - A pack_metadata file for Pack1
+        When
+            - parsing pack metadata files
+        Then
+            - parsing all the data from file successfully
+        """
+        pack = repo.create_pack("Pack1")
+        pack.pack_metadata.write_json(metadata_file_content)
+
+        res = get_pack_metadata_data(pack.pack_metadata.path)
+        result = res.get('Pack1')
+
+        assert 'name' in result.keys()
+        assert result.get('name') == 'Pack1'
+        assert result.get('current_version') == '1.0.0'
+        assert result.get('author') == author
+        assert result.get('certification') == certification
+        assert result.get('tags') == ['Alerts']
+        assert result.get('use_cases') == ['Case Management']
+        assert result.get('categories') == ['Endpoint']
+
+    @staticmethod
+    def test_get_metadata_bad_path(repo, mocker):
```

Comment:
Can you please explain the logic behind this test?

Reply:
done